neo/SmartContract/NeoService.cs

```diff
@@ -165,6 +169,36 @@ public NeoService(TriggerType trigger, Snapshot snapshot)
             #endregion
         }
 
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="engine">r, s, v, messageHash</param>
+        /// <returns></returns>
+        private bool Ecrecover(ExecutionEngine engine)
+        {
+            var r = new System.Numerics.BigInteger(engine.CurrentContext.EvaluationStack.Pop().GetByteArray().Reverse().Concat(new byte[1]).ToArray());
+            var s = new System.Numerics.BigInteger(engine.CurrentContext.EvaluationStack.Pop().GetByteArray().Reverse().Concat(new byte[1]).ToArray());
+            bool v = engine.CurrentContext.EvaluationStack.Pop().GetBoolean();
+            byte[] messageHash = engine.CurrentContext.EvaluationStack.Pop().GetByteArray();
+            try
+            {
+                ECPoint point = ECDsa.KeyRecover(ECCurve.Secp256k1, r, s, messageHash, v, true);
+                engine.CurrentContext.EvaluationStack.Push(point.EncodePoint(false).Skip(1).ToArray());
+            }
+            catch
+            {
+                engine.CurrentContext.EvaluationStack.Push(new byte[] { 0x00 });
```

Comment:
in neo2, false it's empty 
https://github.com/neo-project/neo-vm/blob/30e9d573d1fcad40a21f6ee14d7eb9703a8a6d10/src/neo-vm/Types/Boolean.cs#L9